File: mobile/src/main/java/com/siliconlabs/bledemo/Utils/StringUtils.java
package com.siliconlabs.bledemo.Utils;

public class StringUtils {

    public static String HEX_VALUES = "0123456789abcdefABCDEF";

    public static String getStringWithoutWhitespaces(String text) {
        return text.replaceAll(" ","");
    }

    public static String getStringWithoutColons(String text) {
        if (text == null || text.equals("")) return "";

        StringBuilder result = new StringBuilder();
        String[] splittedStrings = text.split(":");

        for (String str : splittedStrings) {
            result.append(str);
        }

        return result.toString();
    }

}

File: mobile/src/main/java/com/siliconlabs/bledemo/Utils/StringUtils.kt
package com.siliconlabs.bledemo.Utils

object StringUtils {

    const val HEX_VALUES = "0123456789abcdefABCDEF"

    fun getStringWithoutWhitespaces(text: String): String {
        return text.replace(" ".toRegex(), "")
    }

    fun getStringWithoutColons(text: String?): String {
        if (text == null || text == "") return ""
        val result = StringBuilder()
        val splittedStrings = text.split(":".toRegex()).toTypedArray()
        for (str in splittedStrings) {
            result.append(str)
        }
        return result.toString()
    }

    fun removeWhitespaceAndCommaIfNeeded(sb: StringBuilder) {
        val len = sb.length

        if (len >= 2 && sb[len - 1] == ' ' && sb[len - 2] == ',') {
            sb.setLength(len - 2)
        }
    }
}